File: server/hotrod/src/test/scala/org/infinispan/server/hotrod/HotRod1xFunctionalTest.scala
package org.infinispan.server.hotrod

import java.lang.reflect.Method
import java.util.Arrays
import test.HotRodTestingUtil._
import org.testng.Assert._
import org.infinispan.server.hotrod.OperationStatus._

import org.infinispan.server.hotrod.test.{TestResponseWithPrevious, HotRodClient}
import org.testng.annotations.Test

@Test(groups = Array("functional"), testName = "server.hotrod.HotRod1xFunctionalTest")
class HotRod1xFunctionalTest extends HotRodFunctionalTest {
   override protected def connectClient: HotRodClient =
      new HotRodClient("127.0.0.1", hotRodServer.getPort, cacheName, 60, 13)

   override def testSize(m: Method): Unit = {
      // Not supported
   }

   override protected def assertSuccessPrevious(resp: TestResponseWithPrevious, expected: Array[Byte]): Boolean = {
      if (expected == null) assertEquals(None, resp.previous)
      else assertTrue(java.util.Arrays.equals(expected, resp.previous.get))
      assertStatus(resp, Success)
   }

   override protected def assertNotExecutedPrevious(resp: TestResponseWithPrevious, expected: Array[Byte]): Boolean = {
      if (expected == null) assertEquals(None, resp.previous)
      else assertTrue(java.util.Arrays.equals(expected, resp.previous.get))
      assertStatus(resp, OperationNotExecuted)
   }

}

File: server/hotrod/src/test/java/org/infinispan/server/hotrod/HotRod1xFunctionalTest.java
package org.infinispan.server.hotrod;

import static org.infinispan.server.hotrod.OperationStatus.OperationNotExecuted;
import static org.infinispan.server.hotrod.OperationStatus.Success;
import static org.infinispan.server.hotrod.test.HotRodTestingUtil.assertStatus;
import static org.testng.AssertJUnit.assertEquals;
import static org.testng.AssertJUnit.assertTrue;

import java.lang.reflect.Method;
import java.util.Optional;

import org.infinispan.server.hotrod.test.HotRodClient;
import org.infinispan.server.hotrod.test.TestResponseWithPrevious;
import org.testng.annotations.Test;

@Test(groups = "functional", testName = "server.hotrod.HotRod1xFunctionalTest")
public class HotRod1xFunctionalTest extends HotRodFunctionalTest {
   @Override
   protected HotRodClient connectClient() {
      return new HotRodClient("127.0.0.1", hotRodServer.getPort(), cacheName, 60, (byte) 13);
   }

   @Override
   public void testSize(Method m) {
      // Not supported
   }

   @Override
   protected boolean assertSuccessPrevious(TestResponseWithPrevious resp, byte[] expected) {
      if (expected == null)
         assertEquals(Optional.empty(), resp.previous);
      else
         assertTrue(java.util.Arrays.equals(expected, resp.previous.get()));
      return assertStatus(resp, Success);
   }

   @Override
   protected boolean assertNotExecutedPrevious(TestResponseWithPrevious resp, byte[] expected) {
      if (expected == null)
         assertEquals(Optional.empty(), resp.previous);
      else
         assertTrue(java.util.Arrays.equals(expected, resp.previous.get()));
      return assertStatus(resp, OperationNotExecuted);
   }
}
